src/RedditListing.js

```diff
@@ -22,12 +22,10 @@ export default function RedditListing(props) {
     margin: '10px'
   };
 
-  const listTitle = {
+  const titleListItem = {
     margin: '10px'
   };
 
-  const titleListItem = {};
-
   return (
     <li style={listItem}>
       <a style={anchorStyle} href={props.link}>
```

Comment:
So I was trying to do my second PR, but realised it was a bit of refactoring to get it working and you should give it a go yourself! Basically since all your RedditPost-X were the same HTML, the best practice would be to abstract them into their own component.

Reply:
Ah I see what you mean now. But I think in general when people in React talk about "avoiding inheritance" they're mainly talking about extending the class `React.Component`, which provides all the lifecycle methods and state.

Theoretically there's nothing wrong with creating every component that inherits from `React.Component`, practically it might have some perf issues, but we try to avoid that in most cases because lifting state makes it easier to reason about your application's state, attempting to keep it all in one place (https://reactjs.org/docs/lifting-state-up.html) and trying to create more functional components clearly outlines the components which exist only to provide visuals versus those that hold logic (https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0). And you'll start to see why this pattern is desirable when you start making bigger applications as components need to be re-used for their visual aspect more often than components that deal with data. Maybe check out my Hcard Builder for a clearer example of this at a small-medium scale: https://github.com/gyfchong/hcard-builder

Abstraction in this context just means move the code into it's own component so you can re-use it in different places.

... Hope that helps!

Reply:
> A single component and the passed in props would select the CSS, that was my second guess.

Yeah that's the right direction. I would look into implementing CSS modules first, so you can take advantage of conditional classnames (https://github.com/JedWatson/classnames) which will help clean up your code.

Reply:
Awesome thanks so much! 
Def love all the links!